CONTRIBUTING.md

```diff
@@ -9,112 +9,138 @@ There are 4 main ways of contributing to the PyMC project (in descending order o
 * Contributing or improving the documentation (`docs`) or examples (`pymc/examples`)
 * Submitting issues related to bugs or desired enhancements
 
-# Opening issues
+## Opening issues
 
 We appreciate being notified of problems with the existing PyMC code. We prefer that issues be filed the on [Github Issue Tracker](https://github.com/pymc-devs/pymc/issues), rather than on social media or by direct email to the developers.
 
 Please verify that your issue is not being currently addressed by other issues or pull requests by using the GitHub search tool to look for key words in the project issue tracker.
 
 Filter on the ["beginner friendly"](https://github.com/pymc-devs/pymc/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22) label for issues which are good for new contributors.
 
-# Contributing code via pull requests
+## Etiquiette for code contributions
+* When you start working working on an issue, open a `Draft` pull request as soon as you're doing the first steps (see steps below).
+* Before opening a PR with a new feature, please propose it in an [Issue](https://github.com/pymc-devs/pymc/issues) or [Discussion](https://github.com/pymc-devs/pymc/discussions) with the maintainers. Depending on the proposal we might direct you to other places such as [`pymc-experimental`](https://github.com/pymc-devs/pymc-experimental) or [`pymc-examples`](https://github.com/pymc-devs/pymc-examples).
+* Any issue without an open pull request is available for work. However if a pull request has no recent activity it may be closed, or taken over by someone else.
```

Comment:
We should set a time here. What does "recent activity" mean? I have no idea and not knowing if they are supposed to respond within 24h or 3 days to prevent their PR from being closed is very stressful.

Reply:
I didn't want to set a time because it depends a lot on context. If it's a PR to fix a failing CI on `main`, we might take over after a few hours, but if it's a PR adding a new complicated step method or distribution we'll often wait weeks.

It's really a matter of good judgement and I wouldn't add a rule preventing that

Reply:
It doesn't need to be a rule. We know that this will depend on context, and it will also depend on availability, if we are all busy with other things or even other PRs we can leave a PR unchecked for longer independently of context. But many (or most) of the readers of this document/section do not have much/any context so we need to provide that:

* Will they be expected to answer in a matter of hours? No except if very rare exceptions you'll probably never find youself in. There may be PRs where such a fast cadence is needed if they are needed to fix CI on `main` or to fix the documentation website.
* Can they expect to work on a PR only once every 2 weeks? No
* If the PR takes longer than expected and you are required to stop working on it for a while, what happens? Answered already in a point below.

Even expert pymc users might not have any context when it comes to contributing to open source.

Reply:
```suggestion
* Any issue without an open pull request is available for work. However if a pull request has no recent activity it may be closed, or taken over by someone else. What is considered "recent" will depend on the urgency, and everybody is asked to remain calm and use good judgement.
```
Is this better?
Otherwise, please apply another suggestion.

Reply:
How's this?

```suggestion
* Any issue without an open pull request is available for work. 
   * If a pull request has no recent activity it may be closed, or taken over by someone else.
   * The specific timeframe for recent is hard to define as it depends on the contributor the specific code change, and other contextual factors. As a rule of thumb we in a normal pull request with no other blockers there typically is activity every couple of days
   * The core devs will make their best judgement when opting to close PRs
```